argoverse/map_representation/map_api.py

```diff
@@ -378,7 +378,22 @@ def get_ground_height_at_xy(self, point_cloud: np.ndarray, city_name: str) -> np
         npyimage_coords = npyimage_to_city_se2.transform_point_cloud(city_coords)
         npyimage_coords = npyimage_coords.astype(np.int64)
 
-        # index in at (x,y) locations, which are (y,x) in the image
+        # index at (x,y) locations, which are (y,x) in the image
+        max_y = np.max(npyimage_coords[:, 1])
+        max_x = np.max(npyimage_coords[:, 0])
+
+        height_y, height_x = np.shape(ground_height_mat)
+
+        assert np.all(npyimage_coords[:, 1] > 0) and np.all(
+            npyimage_coords[:, 0] > 0
+        ), "Invalid coordinates, please make sure the query location is in a valid city coordinate"
+
+        if max_x > height_x or max_y > height_y:
+            # expand ground height npy image, fill with NaN
+            ground_height_mat_pad = np.full((max_y, max_x), np.nan)
+            ground_height_mat_pad[0:max_y, 0:max_x] = ground_height_mat
+            ground_height_mat = copy.deepcopy(ground_height_mat_pad)
```

Comment:
@janesjanes thanks for fixing the index issue Patsorn.  What is the deepcopy here for? I was using those a bit indiscriminately, i can probably go back and remove some